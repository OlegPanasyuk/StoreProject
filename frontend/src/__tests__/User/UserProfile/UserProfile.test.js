import React from 'react';
import UserProfileConnected, { UserProfile } from '../../../User/UserProfile';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
//import rest from 'rest';

//import { addGoodsToBasket, deleteGoodsFromBasket } from '../../REDUX/actions/actionsShoppingBasket';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    userInfo: {
        username: 'Admin',
        email: 'admin@admin.dog',
        role: 'Admin',
        create_time: '2019-04-22T10:29:35.000Z'
    }
}) {
    let component = shallow(<UserProfile {...props} />);
    return { component, props };
}

describe('UserProfile --- Snapshot', () => {
    it('+++capturing Snapshot of ShoppingBascket', () => {
        const obj = setupComponent();
        expect(UserProfile.propTypes.userInfo).toBe(PropTypes.object);
        expect(shallowToJson(obj.component)).toMatchSnapshot();
    });
});


describe('UserProfile', () => {

    const initialState = {
        userHeaderReducers: {
            userInfo: {
                username: 'Admin',
                email: 'admin@admin.dog',
                role: 'Admin',
                create_time: '2019-04-22T10:29:35.000Z'
            }
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><UserProfileConnected /></Provider>);
    });

    it('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(UserProfileConnected).length).toEqual(1);
    });

});
