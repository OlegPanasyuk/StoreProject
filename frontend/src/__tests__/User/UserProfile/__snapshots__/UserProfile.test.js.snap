// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile --- Snapshot +++capturing Snapshot of ShoppingBascket 1`] = `
<Container
  as="div"
  fluid={false}
>
  <h3>
    User Profile:
  </h3>
  <p>
    <b>
      Name:
    </b>
     
    Admin
    <br />
    <b>
      Email:
    </b>
     
    admin@admin.dog
    <br />
    <b>
      Role:
    </b>
     
    Admin
    <br />
    <b>
      Create Time:
    </b>
     
    2019-04-22T10:29:35.000Z
    <br />
  </p>
</Container>
`;
